logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -146,7 +146,12 @@ private void innerWriteEntry(DLQEntry entry) throws IOException {
 
     @Override
     public synchronized void close() throws IOException {
```

Comment:
I agree that this is something to be looked at, but not as part of this PR. I have more work in this area outstanding in another brach